File: app/src/main/java/com/example/sergejmitrofanov/librarycatalogue/presenter/BookListPresenterImpl.java
package com.example.sergejmitrofanov.librarycatalogue.presenter;

import android.os.Handler;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.example.sergejmitrofanov.librarycatalogue.interactor.BookListUseCase;

public class BookListPresenterImpl implements BookListPresenter {

  @NonNull
  private final BookListUseCase bookListUseCase;
  @Nullable
  private BooksView booksView;

  public BookListPresenterImpl(@NonNull BookListUseCase bookListUseCase) {
    this.bookListUseCase = bookListUseCase;
  }

  public void attachView(@NonNull BooksView booksView) {
    this.booksView = booksView;
    loadBooks();
  }

  public void detachView() {
    this.booksView = null;
  }

  @Override
  public void loadBooks() {
    new Handler().postDelayed(() -> {
      if (booksView != null) {
        booksView.showBooks(bookListUseCase.getBooks());
      }
    }, 2000);

  }
}

File: app/src/main/java/com/example/sergejmitrofanov/librarycatalogue/presenter/BookListPresenterImpl.kt
package com.example.sergejmitrofanov.librarycatalogue.presenter

import android.os.Handler
import android.os.Looper
import com.example.sergejmitrofanov.librarycatalogue.interactor.BookListUseCase
import dagger.hilt.android.scopes.FragmentScoped
import javax.inject.Inject

@FragmentScoped
class BookListPresenterImpl @Inject constructor(
    private val bookListUseCase: BookListUseCase
) : BookListPresenter {
    private var booksView: BooksView? = null

    override fun attachView(booksView: BooksView) {
        this.booksView = booksView
        loadBooks()
    }

    override fun detachView() {
        this.booksView = null
    }

    override fun loadBooks() {
        Handler(Looper.getMainLooper()).postDelayed(
            { booksView?.showBooks(bookListUseCase.getBooks()) },
            2000,
        )
    }
}
